tests/convert_if_to_select/check_error_msgs/invalid_conditionals.mlir

```diff
@@ -0,0 +1,39 @@
+// TODO: Add RUN command and file check for checking diagnostic messages
```

Comment:
See last comment: if we add the hard fail on non-pure ops, this test file can be enabled (see any [invalid-ops.mlir](https://github.com/google/heir/blob/main/tests/arith_ext/invalid-ops.mlir) file in the various test folders for examples for how to do this.)